misk/build.gradle

```diff
@@ -90,68 +89,73 @@ dependencies {
   testCompile project(':misk-testing')
 }
 
-ext.generateDockerContainerName = { service, project, task ->
-  def slug = task.split('/').join('-').split('@').join('')
-  return "${new Date().format("YMD-HMS")}-${service}-${project}-${slug}"
-}
+task web(type: Exec) {
+  // -Pcmd='-d': pass in argument to misk-web shell script in Docker container
+  // -Ptabs='tabs/config, tabs/example': spins up parallel Docker containers with the following paths as volumes
+  // -Pdockerd: boolean to run Docker container as daemon in background
+  // Running webpack-dev-servers for various tabs: ./gradlew web -Pcmd='-d' -Ptabs='tabs/config, tabs/example'
+  // Build all tabs: ./gradlew web
+  // See other misk-web options: ./gradlew web -Pcmd='-h'
+  
+  def generateDockerContainerName = { service, project, task ->
+    def slug = task.split('/').join('-').split('@').join('')
+    return "${new Date().format("YMD-HMS")}-${service}-${project}${slug}"
+  }
 
-ext.getPackageJsonPort = { path ->
-  def packageFile= new File("${project.projectDir}/web/${path}/package.json")
-  def packageJson = new JsonSlurper().parseText(packageFile.text)
-  return packageJson.miskTab.port
-}
+  def getPackageJsonPort = { path ->
+    def packageFile= new File("${project.projectDir}/web/${path}/package.json")
+    def packageJson = new JsonSlurper().parseText(packageFile.text)
+    return packageJson.miskTab.port
+  }
 
-ext.generateDockerRunCommand = { image, service, project, projectPath, path ->
-  def containerName = "${generateDockerContainerName(service, project, path)}"
-  def port = ""
-  if (path.startsWith("tabs")) {
-    port = "-p ${getPackageJsonPort(path)}:${getPackageJsonPort(path)}"
+  def generateDockerRunCommand = { service, project, projectPath, path, cmd, daemon="-d", image = "squareup/misk-web:0.0.1" ->
+    def containerName = "${generateDockerContainerName(service, project, path)}"
+    def port = ""
+    if (path.startsWith("tabs")) {
+      port = "-p ${getPackageJsonPort(path)}:${getPackageJsonPort(path)}"
+    }
+
+    def command = "docker run ${daemon} --rm --name ${containerName} -v ${projectPath}/web${path}:/web${path} ${port} ${image} ${cmd}"
```

Comment:
Ok, I see it is indeed just called by something else. Wrapping is probably not worth if it's never really going to be run 'by hand'.